### PR TITLE
Chap 5 Expand User class and rubocop styling adjustments.

### PR DESCRIPTION
**Pull request by LostSheep**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,6 @@
 
 # Ignore Spring files.
 /spring/*.pid
+
+# Ignore vi bad write file
+.swp
```
Comment (bwang-ith) [inline .gitignore]: This isn't going to ignore anything except files literally named .swp
Use a wildcard: `.swp` -> `*.swp`

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+/* variables */
+
+$light-gray: #777;
+
+/* universal */
+
+body {
+  padding-top: 60px;
+}
```
Comment (bwang-ith) [inline app/assets/stylesheets/custom.scss:12]: add blank lines between all top level styles. Do not have blank lines between nested styles.

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -10,4 +10,7 @@ def help; end
 
   # GET static_pages/about.
   def about; end
+
+  # GET static_pages/contact.
```
Comment (bwang-ith) [inline app/controllers/static_pages_controller.rb]: `static_pages` -> `/static_pages`, Also, you need the second "human" comment for these controller actions. Same for whole file.

File: app/views/layouts/_footer.html.erb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<footer class='footer'>
+  <small>
+    The <a href='hrrp://www.railstutorial.org/'>Ruby on Rails Tutorial</a>
+    by <a href='http://www.michaelharl.com/'>Michael Hartl</a>
+  </small>
+  <nav>
+    <ul>
+      <li><%= link_to 'About',    about_path %></li>
+      <li><%= link_to 'Contact',  contact_path %></li>
```
Comment (bwang-ith) [inline app/views/layouts/_footer.html.erb]: I like the alignment between the two lines. However you've got an extra space here in the 'Contact' line that you don't need.

File: app/views/layouts/_header.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", '#', id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",   home_path %></li>
```
Comment (bwang-ith) [inline app/views/layouts/_header.html.erb]: same

File: app/views/layouts/_header.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        <li><%= link_to "Help",   help_path %></li>
+	<li><%= link_to "Log in", '#' %></li>
```
Comment (bwang-ith) [inline app/views/layouts/_header.html.erb]: fix indent

File: test/controllers/microposts_controller_test.rb [hunk 7]
```diff
@@ -6,9 +6,9 @@
 class MicropostsControllerTest < ActionDispatch::IntegrationTest
   # Setup for micropost tests.
   setup do
-    @micropost = microposts(:one)
-    @user = users(:one)
-    @base_title = Contant::PAGE_TITLE 
+    @micropost  = microposts(:one)
```
Comment (bwang-ith) [inline test/controllers/microposts_controller_test.rb:9]: nice

File: test/controllers/microposts_controller_test.rb [hunk 8]
```diff
@@ -46,15 +46,16 @@ class MicropostsControllerTest < ActionDispatch::IntegrationTest
...
-  test 'should succeed patch update micropost' do
-    patch micropost_path(@micropost), params: 
-    { 
-      micropost: 
-      { 
-        content: @micropost.content, 
-        user_id: @micropost.user_id
-      }, format: :html
-    }
+  test 'should succeed update micropost' do
```
Comment (bwang-ith) [inline test/controllers/microposts_controller_test.rb]: you changed the test name. It met our convention and now it does not.

File: test/controllers/static_pages_controller_test.rb [hunk 9]
```diff
@@ -4,7 +4,7 @@
 
 # Tests static pages controler.
 class StaticPagesControllerTest < ActionDispatch::IntegrationTest
-  # Test setup
+  # Test setup.
```
Comment (bwang-ith) [inline test/controllers/static_pages_controller_test.rb]: nice

File: test/controllers/static_pages_controller_test.rb [hunk 10]
```diff
@@ -19,24 +19,30 @@ def setup
...
 
   # Test about.
 
   test 'should succeed get about' do
-    get static_pages_about_path
+    get about_path
     assert_response :success
     assert_select 'title', "About | #{@base_title}"
   end
+
```
Comment (bwang-ith) [inline test/controllers/static_pages_controller_test.rb:42]: missing section comment

File: test/controllers/users_controller_test.rb [hunk 11]
```diff
@@ -2,17 +2,19 @@
 
 require 'test_helper'
 
-# Tests the user controll:er.
+# Tests the user controller.
```
Comment (bwang-ith) [inline test/controllers/users_controller_test.rb]: sentence

**Review by bwang-ith — CHANGES_REQUESTED** [review]

File: app/views/layouts/_header.html.erb (hunk 5 above)
Comment (LostSheep) [reply app/views/layouts/_header.html.erb]: In the header they are lined up with login.

File: app/views/layouts/_header.html.erb (hunk 6 above)
Comment (LostSheep) [reply app/views/layouts/_header.html.erb]: vim didn't show a difference but turns out there was a tab mixed in with the spaces.

File: app/views/layouts/_header.html.erb (hunk 5 above)
Comment (LostSheep) [reply app/views/layouts/_header.html.erb]: After the correction below home_path and help_path should be lined up with the destination link of Log in.

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+footer {
+  margin-top: 45px;
+  padding-top: 5px;
+  border-top: 1px solid #eaeaea;
+  color: $light-gray;
+
+  a {
+    color: #555;
+
```
Comment (bwang-ith) [inline app/assets/stylesheets/custom.scss]: remove blank line

File: app/assets/stylesheets/custom.scss [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  small {
+    float: left;
+  }
+
+  ul {
+    float: right;
+    list-style: none;
+
```
Comment (bwang-ith) [inline app/assets/stylesheets/custom.scss]: same

File: app/assets/stylesheets/static_pages.scss [hunk 14]
```diff
@@ -1,3 +1,35 @@
...
+#guilt-signup {
+  margin: auto;
+  position: relative;
+  width: 640px;
+  .signup-container {
+    bottom: 5px;
+    margin: 0px calc(50% - (#{$signup-width}/2));
+    position: absolute;
+    width: $signup-width;
+
```
Comment (bwang-ith) [inline app/assets/stylesheets/static_pages.scss]: same. Remember, no blank lines in nested styles.

File: app/controllers/static_pages_controller.rb [hunk 15]
```diff
@@ -5,9 +5,12 @@ class StaticPagesController < ApplicationController
   # GET static_pages/home.
   def home; end
 
-  # GET static_pages/help.
+  # GET /static_pages/help.
```
Comment (bwang-ith) [inline app/controllers/static_pages_controller.rb:9]: still need extra "human" comment. Remember, controller actions get both.

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -0,0 +1,12 @@
+<header class="navbar navbar-fixed-top navbar-inverse">
+  <div class="container">
+    <%= link_to "sample app", '#', id: "logo" %>
+    <nav>
+      <ul class="nav navbar-nav navbar-right">
+        <li><%= link_to "Home",   home_path %></li>
+        <li><%= link_to "Help",   help_path %></li>
```
Comment (bwang-ith) [inline app/views/layouts/_header.html.erb]: use single quotes

File: test/controllers/microposts_controller_test.rb [hunk 17]
```diff
@@ -47,14 +47,15 @@ class MicropostsControllerTest < ActionDispatch::IntegrationTest
...
-      }, format: :html
-    }
+    patch micropost_path(@micropost),
+          params:
+          {
+            micropost:
+            {
+              content: @micropost.content,
+              user_id: @micropost.user_id
+            }, format: :html
```
Comment (bwang-ith) [inline test/controllers/microposts_controller_test.rb]: move `format: :html` to the same indent level as `Micropost:`. On second thought, you can probably just delete it. Remember, we figured out it was a different bug.

File: test/controllers/users_controller_test.rb [hunk 18]
```diff
@@ -2,17 +2,20 @@
 
 require 'test_helper'
 
-# Tests the user controll:er.
+# Tests user controller.
```
Comment (bwang-ith) [inline test/controllers/users_controller_test.rb]: Make it a sentence. e.g. `# Test the user controller.`

**Review by bwang-ith — CHANGES_REQUESTED** [review]

**Review by bwang-ith — APPROVED** [review]

**Pull request merged**